### PR TITLE
adds space slug script to the migration process

### PR DESCRIPTION
**Pull request by denisdimitrov**
- Adds the space slug script to the migration

### CODE REVIEW

File: scripts/migration/v0.0.5__add-space-slug.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      };
+
+      await space.ref.set(spaceWithSlug);
+      console.log(
+        `The slug: '${slug}' has been successfully added for venue: ${spaceData.name}`
+      );
+    }
+  } catch (error) {
+    console.log("The script failed to update all of the spaces. Error:");
+    console.log(error);
```
Comment (goran-peoski-work) [inline scripts/migration/v0.0.5__add-space-slug.ts:76]: no `return` here means the following code will also be executed, meaning two messages, an error followed by a success one

**Review by goran-peoski-work — CHANGES_REQUESTED** [review]
A minor error logging nitpick.

Also, wasn't there another system for migrations that didn't require all this old scripts' boilerplate?

File: scripts/migration/v0.0.5__add-space-slug.ts (hunk 1 above)
Comment (denisdimitrov) [reply scripts/migration/v0.0.5__add-space-slug.ts:46]: `return` added and the script has been refactored to the new syntax in: https://github.com/sparkletown/sparkle/pull/2584/commits/ebd1d7da0e1937aea6bdcdc332f5b4eb662679b9

**Review by goran-peoski-work — APPROVED** [review]
LGTM.

For the future I can recommend using `assert` from Node.js and the `chalk` library. Example: https://github.com/sparkletown/sparkle/blob/staging/scripts/simulator.ts#L64

**Comment by ermiaSparkle** [comment]
@sunny-viktoryia would u please check this one
We need to know it and run it as migrations steps.

File: scripts/migration/v0.0.5__add-space-slug.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      if (!spaceData.name) {
+        console.log(
+          `Skipping space, name is missing and slug couldn't be generated. Space id: ${space.id}, space's world id: ${spaceData.worldId}.`
+        );
+        continue;
+      }
+
+      // If the space already has a slug, skip the space because it will override it.
+      if (spaceData.slug) {
+        console.log(`Skipping space ${spaceData.name}. Slug already exists`);
```
Comment (sunny-viktoryia) [inline scripts/migration/v0.0.5__add-space-slug.ts:28]: can we include space.id in the message too? 
```suggestion
        console.log(`Skipping space ${spaceData.name} (${space.id}). Slug already exists`);
```

File: scripts/migration/v0.0.5__add-space-slug.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+      }
+
+      const slug = generateSlug(spaceData.name);
+
+      const spaceWithSlug = {
+        ...spaceData,
+        slug,
+      };
+
+      await space.ref.set(spaceWithSlug);
```
Comment (sunny-viktoryia) [inline scripts/migration/v0.0.5__add-space-slug.ts:39]: won't it work this way?
```suggestion
      await space.ref.update({ slug });
```

File: scripts/migration/v0.0.5__add-space-slug.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+export const migrate = async ({ firestore }: MigrateOptions) => {
+  console.log(`Fetching spaces...`);
+
+  const spaces = await firestore.collection("venues").get();
+
+  console.log(`Updating spaces...`);
+
+  try {
+    for (const space of spaces.docs) {
```
Comment (sunny-viktoryia) [inline scripts/migration/v0.0.5__add-space-slug.ts:15]: nit:
```suggestion
  const { docs: spaces } = await firestore.collection("venues").get();

  console.log(`Updating spaces...`);

  try {
    for (const space of spaces) {
```

**Review by sunny-viktoryia — APPROVED** [review]
the script looks straightforward, thanks! ❤️ 
few minor comments, you can address them if you like.

File: scripts/migration/v0.0.5__add-space-slug.ts (hunk 4 above)
Comment (denisdimitrov) [reply scripts/migration/v0.0.5__add-space-slug.ts]: Fixed in: https://github.com/sparkletown/sparkle/pull/2584/commits/951fc2b03c6f9ae842867d9473c64c2d759ed03f

File: scripts/migration/v0.0.5__add-space-slug.ts (hunk 3 above)
Comment (denisdimitrov) [reply scripts/migration/v0.0.5__add-space-slug.ts]: Fixed in: https://github.com/sparkletown/sparkle/pull/2584/commits/951fc2b03c6f9ae842867d9473c64c2d759ed03f

File: scripts/migration/v0.0.5__add-space-slug.ts (hunk 2 above)
Comment (denisdimitrov) [reply scripts/migration/v0.0.5__add-space-slug.ts]: Fixed in: https://github.com/sparkletown/sparkle/pull/2584/commits/951fc2b03c6f9ae842867d9473c64c2d759ed03f

**Pull request merged**